packages/sdk/src/chains/BaseAssetTransfer.ts

```diff
@@ -45,6 +62,70 @@ export abstract class BaseAssetTransfer {
       resource: resource,
     };
 
+    if (destinationProviderUrl) {
+      const destinationHandlerBalance = await this.fetchDestinationHandlerBalance(
+        destinationProviderUrl,
+        transfer,
+      );
+      transfer.details.destinationHandlerBalance = BigInt(destinationHandlerBalance.toString());
```

Comment:
Just throw here instead and remove everything in the asset transfer classes.

Reply:
Throwing an error here does not allow us inform the user of what the maximum balance they can transfer is.

See the discussion [here](https://chainsafeio.slack.com/archives/C02KNM76NR4/p1690995062352929)

Reply:
It is more elegant to throw a custom error with the data of the error than to paste that data into the transfer.

Reply:
If you create a LiquidityError and then inside add maximum transfer amount you would achieve the same without needing to add extra logic

Reply:
Throwing an error also goes against the original requirement of making this check default but optional - which was one of the original requirements of the ticket.

Reply:
But you throw an error anyway in the assetTransfer class.
It is optional in the sense that if you don't provide the destination RPC it won't trigger.
But if there is no liquidity then it is not optional to throw an error.